packages/react-theme-provider/src/components/ThemeProvider/useThemeProvider.ts

```diff
@@ -37,18 +38,8 @@ export const useThemeProvider = (
   const localTheme = state.theme;
 
   state.theme = mergeThemes(parentTheme, localTheme);
-  state.style = React.useMemo(() => {
-    // TODO: should we consider insertion to head?
-    //       - how to modify, remove styles?
-    //       - SSR rendering
-
-    // TODO: what variables should be rendered? Merged or only changed?
-    // TODO: how we will proceed with Portals?
-    return {
-      ...state.style,
-      ...themeToCSSVariables(state.theme),
-    };
-  }, [state.style, state.theme]);
+  const themeClassName = useThemeStyleTag(state.theme, state.document);
```

Comment:
From where comes `state.document`? From a React context or from props?

Reply:
props, `FluentProvider` currently already passes (spreads) its document into `ThemeProvider`

Reply:
In this case it should be added to `ThemeProviderProps`:

```diff
export interface ThemeProviderProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
+ document?: Document;
  theme?: PartialTheme | Theme;
}
```

![image](https://user-images.githubusercontent.com/14183168/114521017-50110780-9c42-11eb-871e-42cc447290b3.png)

Reply:
Updated for `ThemeProvider` will create an item in 42 board for `FluentProvider` 

will probably do it myself but in another PR

Reply:
I don't like the `document` has been added to `ThemeProvider` and not to `FluentProvider`.
Do I understand the last comment correctly that you plan to move it?

Reply:
`document` is already in `FluentProvider` the discussion here is to rename the variable so it does not clash with scope